examples/2d/pixel_grid_snap.rs

```diff
@@ -0,0 +1,158 @@
+    // the sample sprite that will be rendered to the pixel-perfect canvas
+    commands.spawn((
+        SpriteBundle {
+            texture: asset_server.load("pixel/bevy_pixel_dark.png"),
+            transform: Transform::from_xyz(-40., 20., 2.),
+            ..default()
+        },
+        Drawable,
+        RenderLayers::layer(1),
+    ));
+
+    // the sample sprite that will be rendered to the high-res "outer world"
+    commands.spawn((
+        SpriteBundle {
+            texture: asset_server.load("pixel/bevy_pixel_light.png"),
+            transform: Transform::from_xyz(-40., -20., 2.),
+            ..default()
+        },
+        Drawable,
+    ));
+}
+
+fn setup_mesh(
+    mut commands: Commands,
+    mut meshes: ResMut<Assets<Mesh>>,
+    mut materials: ResMut<Assets<ColorMaterial>>,
+) {
+    commands.spawn((
+        MaterialMesh2dBundle {
+            mesh: meshes.add(Mesh::from(shape::Quad::default())).into(),
+            transform: Transform::from_xyz(40., 0., 2.).with_scale(Vec3::splat(32.)),
+            material: materials.add(ColorMaterial::from(Color::BLACK)),
+            ..default()
+        },
+        RenderLayers::layer(1),
+        Drawable,
+    ));
+}
+
+fn setup_camera(mut commands: Commands, mut images: ResMut<Assets<Image>>) {
+    let canvas_size = Extent3d {
+        width: RES_WIDTH,
+        height: RES_HEIGHT,
+        ..default()
+    };
+
+    // this Image serves as a canvas representing the low-resolution game screen
+    let mut canvas = Image {
+        texture_descriptor: TextureDescriptor {
+            label: None,
+            size: canvas_size,
+            dimension: TextureDimension::D2,
+            format: TextureFormat::Bgra8UnormSrgb,
+            mip_level_count: 1,
+            sample_count: 1,
+            usage: TextureUsages::TEXTURE_BINDING
+                | TextureUsages::COPY_DST
+                | TextureUsages::RENDER_ATTACHMENT,
+            view_formats: &[],
+        },
+        ..default()
+    };
+
+    // fill image.data with zeroes
+    canvas.resize(canvas_size);
+
+    let image_handle = images.add(canvas);
+
+    // this camera renders whatever is on Render layer 1 to the canvas
+    commands.spawn((
+        Camera2dBundle {
+            camera: Camera {
+                // render before the "main pass" camera
+                order: -1,
+                target: RenderTarget::Image(image_handle.clone()),
+                ..default()
+            },
+            ..default()
+        },
+        RenderLayers::layer(1),
+        InGameCamera,
+    ));
+
+    // spawn the canvas
+    commands.spawn((
+        SpriteBundle {
+            texture: image_handle,
+            ..default()
+        },
+        Canvas,
+    ));
+
+    // the "outer" camera renders the objects that are not rendered to the canvas
+    // here, the canvas and one of the sample sprites will be rendered by this camera
+    commands.spawn((Camera2dBundle::default(), OuterCamera));
```

Comment:
Just an opinion, but I feel that `RenderLayers::layer(1)` should be moved into these two entities instead.

The idea is that most entities we're spawning is going to go into the low-res canvas. By doing it this way we don't need to tag every low-res entities with `RenderLayers::layer(1)`.

Reply:
Agree, made the `RenderLayers` usage more detailed.